test/functional/custom-decorators.spec.ts

```diff
@@ -1,261 +1,241 @@
-            });
+    it("if firstName is empty and lastLame is not empty then it should fail", () => {
+        expect.assertions(2);
+        const model = new SecondClass();
+        model.firstName = "";
+        model.lastName = "Kim";
+        return validator.validate(model).then(errors => {
+            expect(errors.length).toEqual(1);
+            expect(errors[0].constraints).toEqual({isLonger: "firstName must be longer then lastName"});
         });
+    });
 
-        it("if firstName is shorter then lastLame then it should fail", function() {
-            const model = new SecondClass();
-            model.firstName = "Li";
-            model.lastName = "Kim";
-            return validator.validate(model).then(errors => {
-                errors.length.should.be.equal(1);
-                errors[0].constraints.should.be.eql({ isLonger: "firstName must be longer then lastName" });
-            });
+    it("if firstName is shorter then lastLame then it should fail", () => {
+        expect.assertions(2);
+        const model = new SecondClass();
+        model.firstName = "Li";
+        model.lastName = "Kim";
+        return validator.validate(model).then(errors => {
+            expect(errors.length).toEqual(1);
+            expect(errors[0].constraints).toEqual({isLonger: "firstName must be longer then lastName"});
         });
-
     });
+});
 
-    describe("decorator with separate validation constraint class", function() {
-
-        @ValidatorConstraint({ name: "isShortenThan" })
-        class IsShortenThanConstraint implements ValidatorConstraintInterface {
-
-            validate(value: any, args: ValidationArguments) {
-                const [relatedPropertyName] = args.constraints;
-                const relatedValue = (args.object as any)[relatedPropertyName];
-                if (value === null || value === undefined)
-                    return true;
-
-                return  typeof value === "string" &&
-                        typeof relatedValue === "string" &&
-                        value.length < relatedValue.length;
-            }
-
-        }
-
-        function IsShortenThan(property: string, validationOptions?: ValidationOptions) {
-            return function (object: Object, propertyName: string) {
-                registerDecorator({
-                    target: object.constructor,
-                    propertyName: propertyName,
-                    options: validationOptions,
-                    constraints: [property],
-                    validator: IsShortenThanConstraint
-                });
-            };
-        }
-
-        class MyClass {
-            firstName: string;
-
-            @IsShortenThan("firstName", {
-                message: "$property must be shorter then $constraint1. Given value: $value"
-            })
-            lastName: string;
+describe("decorator with separate validation constraint class", function () {
```

Comment:
Please use everywhere arrow functions `() => {` :)

Reply:
Yes, my mistake. I thought I had changed them all but missed this one.